src/main/java/org/jenkinsci/maven/plugins/hpi/WarMojo.java

```diff
@@ -58,7 +67,18 @@ public void execute() throws MojoExecutionException {
             rezip.addZipfileset(z);
 
             getProject().setArtifacts(resolveDependencies(dependencyResolution));
-            for( MavenArtifact a : getProjectArtifacts() ) {
+
+            Set<MavenArtifact> projectArtifacts = new LinkedHashSet<>(getProjectArtifacts());
+            if(addThisPluginToCustomWar) {
```

Comment:
Would be nice to ensure that the artifact actually has the `hpi` packaging.

Reply:
added packaging type check in commit [de594d9](https://github.com/jenkinsci/maven-hpi-plugin/pull/86/commits/de594d96b895b696a327dd231866d6159b48f9e0)

Reply:
thanks!